modules/generic/src/main/scala/pureconfig/generic/semiauto.scala

```diff
@@ -13,4 +13,32 @@ object semiauto {
 
   final def deriveConvert[A](implicit reader: Lazy[DerivedConfigReader[A]], writer: Lazy[DerivedConfigWriter[A]]): ConfigConvert[A] =
     ConfigConvert.fromReaderAndWriter(Derivation.Successful(reader.value), Derivation.Successful(writer.value))
+
+  final def deriveEnumerationReader[A](
+    implicit
+    readerBuilder: Lazy[EnumerationConfigReaderBuilder[A]]): ConfigReader[A] =
+    deriveEnumerationReader(_.toLowerCase)
```

Comment:
That lowercase default is an annoyance to me. I'd forgotten about it last week and spent a few minutes wondering why nothing worked as I expected.

I don't love the idea of a breaking change but moving in the direction of a more consistent API with less surprising defaults seems like a good reason for it.

Reply:
Thinking about it more, if we're going to make a change change to the naming conventions for enums, I think doing it when this API is introduced would be preferable to waiting until afterward and then breaking this API. My vote is that if we're going to switch, we should do in this PR.

Reply:
I also prefer having pascal case to kebab case as the default transformation. I agree with @leifwickland that this a good opportunity to do it. I've done it in 2fdbf7a.

I'm not sure of how to put this in the changelog, though. Should we put it as a breaking change, or not mention it at all, since it's the behavior of a new method?

Reply:
I suggest that the changelog mention that the old way is deprecated and mention in the new features section that the new approach has a different naming behavior than the old.

Reply:
@jcazevedo if we are to change the lowercase default it should not be only for enumeration readers, but for other coproduct hints too - otherwise we'll be encoding the same data (class names) differently in different hints, which is worse than being inconsistent between product and coproduct hints. Doing that requires "real" breaking changes, though :)

Reply:
I'm OK with introducing that breaking change for the next version. I suppose we can improve the `NoValidCoproductChoiceFound` failure to show more information when deriving a reader with `FieldCoproductHint`. Showing the value found in the discriminating field together with a mention of the changed behavior should be helpful for users to migrate.

Reply:
Cool! Can you do that change on this PR, to avoid having two different conventions when we merge this? :)

Reply:
Sure. I'll update this tomorrow.

Reply:
I changed the default transformation of `FieldCoproductHint` to `PascalCase` -> `kebab-case` in 1735d05. 

I introduced a new `UnexpectedValueForFieldCoproductHint` failure reason, in order to improve error messages when using `FieldCoproductHint`. Incidentally, I moved the responsibility of raising `ConfigReaderFailures` when the disambiguation fails to `CoproductHint`. I think this also adds some extra flexibility to implementations of `CoproductHint`, but let me know what you think.

Sorry for the delay!